src/components/KeyboardSpacer/BaseKeyboardSpacer.js

```diff
@@ -1,69 +1,51 @@
-import React, {PureComponent} from 'react';
+import React, {useState, useEffect, useCallback} from 'react';
 import {Dimensions, Keyboard, View} from 'react-native';
 import * as StyleUtils from '../../styles/StyleUtils';
 import {propTypes, defaultProps} from './BaseKeyboardSpacerPropTypes';
 
-class BaseKeyboardSpacer extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = {
-            keyboardSpace: 0,
-        };
-        this.keyboardListeners = null;
-        this.updateKeyboardSpace = this.updateKeyboardSpace.bind(this);
-        this.resetKeyboardSpace = this.resetKeyboardSpace.bind(this);
-    }
-
-    componentDidMount() {
-        const updateListener = this.props.keyboardShowMethod;
-        const resetListener = this.props.keyboardHideMethod;
-        this.keyboardListeners = [Keyboard.addListener(updateListener, this.updateKeyboardSpace), Keyboard.addListener(resetListener, this.resetKeyboardSpace)];
-    }
-
-    componentWillUnmount() {
-        this.keyboardListeners.forEach((listener) => listener.remove());
-    }
+function BaseKeyboardSpacer(props) {
+    const [keyboardSpace, setKeyboardSpace] = useState(0);
 
     /**
      * Update the height of Keyboard View.
      *
      * @param {Object} [event] - A Keyboard Event.
      */
-    updateKeyboardSpace(event) {
+    const updateKeyboardSpace = useCallback((event) => {
         if (!event.endCoordinates) {
             return;
         }
 
         const screenHeight = Dimensions.get('window').height;
-        const keyboardSpace = screenHeight - event.endCoordinates.screenY + this.props.topSpacing;
-        this.setState(
-            {
-                keyboardSpace,
-            },
-            this.props.onToggle(true, keyboardSpace),
-        );
-    }
+        const space = screenHeight - event.endCoordinates.screenY + props.topSpacing;
+        setKeyboardSpace(space);
+        props.onToggle(true, keyboardSpace);
```

Comment:
![Screenshot 2023-07-18 at 23 05 08](https://github.com/Expensify/App/assets/25520267/3214b53c-111b-4587-8967-7d326f908483)

The next value of `keyboardSpace` is not yet available and still has the previous state value. 

```suggestion
        props.onToggle(true, space);
```